docs/developers_notes/basis_module.md

```diff
@@ -0,0 +1,130 @@
+4. Calls the `_evaluate` method on the input, which is the subclass-specific implementation of the basis set evaluation.
+5. Returns a numpy array of dimension `N x number of bases`, with each basis element evaluated at the samples.
+
+### The Public Method `evaluate_on_grid`
+
+The `evaluate_on_grid` method evaluates the basis set on a grid of equidistant sample points. The user specifies the input as a series of integers, one for each dimension of the basis function, that indicate the number of sample points in each coordinate of the grid.
+
+This method performs the following steps:
+
+1. Checks that the number of inputs matches what the basis being evaluated expects (e.g., one input for a 1-D basis, N inputs for an N-D basis, or the sum of N 1-D bases), and raises a `ValueError` if this is not the case.
+2. Checks that the output of the method does not exceed `gb_limit` GB of memory size. This size limit is specified at class initialization.
+3. Calls the subclass-specific `_get_samples` method, which returns equidistant samples over the domain of the basis function. The domain may depend on the type of basis.
+4. Calls the subclass-specific `_evaluate` method.
+5. Returns both the sample grid points and the evaluation output at each grid point.
+
+### Abstract Methods
+
+The `neurostatslib.basis.Basis` class has the following abstract methods, which every non-abstract subclass must implement:
+
+1. `_evaluate`: Evaluates a basis over some specified samples.
+2. `_get_samples`: Returns a tuple of equidistant samples (as flat numpy arrays) for each dimension of the basis function. These samples are used to form a grid over which the basis is evaluated.
+3. `_check_n_basis_min`: Checks the minimum number of basis functions required. This requirement can be specific to the type of basis.
+
+
+## The `AdditiveBasis` Class
+
+This class extends the abstract `Basis` class and represents a set of basis functions that are combined additively. It overrides the `_evaluate`, `_get_samples`, and `_check_n_basis_min` methods with specific implementations for additive basis functions.
+
+### Overridden Methods
+
+1. `_evaluate`: For an `AdditiveBasis` object, this method should evaluate each basis function at the specified samples and then add the results together.
+2. `_get_samples`: This method should return equidistant samples over the domain of each basis function. The samples should then be summed together.
+3. `_check_n_basis_min`: This method checks whether the number of basis functions meets the minimum requirement for an additive basis set.
+
+## The `MultiplicativeBasis` Class
+
+This class extends the `Basis` abstract class and represents a set of basis functions that are combined multiplicatively. It overrides the `_evaluate`, `_get_samples`, and `_check_n_basis_min` methods with specific implementations for multiplicative basis functions.
```

Comment:
needs a longer description of what it means to combine bases multiplicatively or a link to an example